src/Components/test/testassets/BasicTestApp/Program.cs

```diff
@@ -61,5 +66,19 @@ private static async Task SimulateErrorsIfNeededForTest()
                 throw new InvalidTimeZoneException("This is an asynchronous startup exception");
             }
         }
+
+        private class ConfigureCorsHandler : DelegatingHandler
+        {
+            public ConfigureCorsHandler(HttpMessageHandler instance)
+                : base(instance)
+            {
+            }
+
+            protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                request.WithRequestCredentials(RequestCredentials.Include);
+                return base.SendAsync(request, cancellationToken);
```

Comment:
Seeing the usage like this makes me think all the methods should be called `SetXyz` instead of `WithXyz`.

The reason is that `WithXyz` looks such a lot like the familiar clone-with-alteration pattern used with immutable objects. It looks strange here that we're relying on `WithRequestCredentials` actually mutating the supplied instance.

If the code instead was:

```
request.SetRequestCredentials(RequestCredentials.Include);
```

... then it would be much clearer that yes, we are mutating the instance.

What do you think?

Reply:
I have no objections to calling it either, but `Set*` sounds fine.